lib/internal/http2/compat.js

```diff
@@ -722,6 +717,14 @@ class Http2ServerResponse extends Stream {
       encoding = 'utf8';
     }
 
+    if ((state.closed || state.ending) &&
+        state.headRequest === stream.headRequest) {
+      if (cb) {
```

Comment:
To be alligned with the code below.
```suggestion
      if (typeof cb === 'function') {
```